doc/guide/sample_ml_app.hpp

```diff
@@ -0,0 +1,194 @@
+#include "mlpack/methods/random_forest/random_forest.hpp"
+#include "mlpack/methods/decision_tree/random_dimension_select.hpp"
+#include "mlpack/core/cv/k_fold_cv.hpp"
+#include "mlpack/core/cv/metrics/accuracy.hpp"
+#include "mlpack/core/cv/metrics/precision.hpp"
+#include "mlpack/core/cv/metrics/recall.hpp"
+#include "mlpack/core/cv/metrics/F1.hpp"
+@endcode
+
+Also, we will use the following namespaces:
+
+@code
+using namespace arma;
+using namespace mlpack;
+using namespace mlpack::tree;
+using namespace mlpack::cv;
+@endcode
+
+@section sample_load_dataset Loading the dataset
+
+First step is about loading the dataset. Here we load a CSV dataset 
+(i.e.: mlpack/tests/data/german.csv), assuming the labels don't require normalization.
+
+@code
+bool loaded = mlpack::data::Load("data/german.csv", dataset);
+if (!loaded)
+	return -1;
+@endcode
+
+Then we need to extract the labels from the last dimension of the dataset and remove the
+labels from the training set:
+
+@code
+Row<size_t> labels;
+labels = conv_to<Row<size_t>>::from(dataset.row(dataset.n_rows - 1));
+dataset.shed_row(dataset.n_rows - 1);
+@endcode
+
+We now have our dataset ready for training.
+
+@section sample_training Training
+
+This app will use a Random Forest classifier. At first we define the classifier parameters and then
+we create the classifier to train it.
+
+@code
+const size_t numClasses = 2;
+const size_t minimumLeafSize = 5;
+const size_t numTrees = 10;
+
+RandomForest<GiniGain, RandomDimensionSelect> rf;
+
+rf = RandomForest<GiniGain, RandomDimensionSelect>(dataset, labels,
+	numClasses, numTrees, minimumLeafSize);
+@endcode
+
+Now that the training is completed, we quickly compute the training accuracy:
+
+@code
+Row<size_t> predictions;
+rf.Classify(dataset, predictions);
+const size_t correct = arma::accu(predictions == labels);
+printf("\nTraining Accuracy: %f", (double(correct) / double(labels.n_elem)));
+@endcode
+
+@section sample_crossvalidation Cross-Validating
+
+To evaluate the classifier, we use K-Fold cross-validation. We also define which metric to use in order
+to assess the quality of the trained model.
+
+@code
+const size_t k = 10;
+KFoldCV<RandomForest<GiniGain, RandomDimensionSelect>, Accuracy> cv(k, 
+	dataset, labels, numClasses);
+double cvAcc = cv.Evaluate(numTrees, minimumLeafSize);
+printf("\nKFoldCV Accuracy: %f", cvAcc);
+@endcode
+
+To compute other relevant metrics, such as Precision, Recall and F1:
+
+@code
+double cvPrecision = Precision<Binary>::Evaluate(rf, dataset, labels);
+printf("\nPrecision: %f", cvPrecision);
+
+double cvRecall = Recall<Binary>::Evaluate(rf, dataset, labels);
+printf("\nRecall: %f", cvRecall);
+
+double cvF1 = F1<Binary>::Evaluate(rf, dataset, labels);
+printf("\nF1: %f", cvF1);
+@endcode
+
+@section sample_save_model Saving the model
+
+Now that our model is trained and validated, we save it to a file so we can use it later.
+
+@code
+mlpack::data::Save("mymodel.xml", "model", rf, false,
```

Comment:
Save should be able to derive the format from the filename.

Reply:
Right, removed the unnecessary parameter